artifact/artifact/v1alpha/artifact.proto

```diff
@@ -419,6 +419,9 @@ message File {
   int32 total_chunks = 15 [(google.api.field_behavior) = OUTPUT_ONLY];
   // total tokens
   int32 total_tokens = 16 [(google.api.field_behavior) = OUTPUT_ONLY];
+
+  // Custom metadata provided by the user during file upload
+  optional google.protobuf.Struct external_metadata = 17 [(google.api.field_behavior) = OPTIONAL];
```

Comment:
❓ In the future, when the Console needs to know the source of the file, does it mean the Console needs to parse metadata to get the key like `source` with the current design?

If so, do we need something like `source` enum as `read-only` field as a single source of truth for this field?

Reply:
The console does not need to parse the data; the backend can handle it and populate corresponding response.

we might need the enum but I don't want to overkill right now. we can have that when we have real user flow and product design.